test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -4208,46 +4241,77 @@ def _load_configuration(self, infile, objtype=None):
-
+                try:
+                    conf = json.load(f)
+                except ValueError:
+                    self.logger.info("Error loading JSON file: %s"
+                                     % infile)
+            qmgr = os.path.join(self.client_conf['PBS_EXEC'],
+                                'bin', 'qmgr')
             for k, v in conf.items():
-                fn = self.du.create_temp_file()
-                with open(fn, 'w') as fd:
-                    # handle server data saved as output of qmgr commands
-                    # by piping data back into qmgr
-                    if k.startswith('qmgr_'):
-                        qmgr = os.path.join(self.client_conf['PBS_EXEC'],
-                                            'bin', 'qmgr')
-                        fd.write("\n".join(v))
-                        self.du.run_cmd(self.hostname, [qmgr], cstdin=fd,
-                                        sudo=True)
-                    else:
-                        fd.write("\n".join(v))
-                        # append the last line
-                        fd.write("\n")
-                        self.du.run_cmd(self.hostname, ['cp', fn, k],
-                                        sudo=True)
-                    os.remove(fn)
-
+                # Load server configuration
+                if k.startswith('qmgr_'):
+                    fpath = self.du.create_temp_file()
+                    print_svr = '\n'.join(v)
+                    with open(fpath, 'w') as f:
+                        f.write(print_svr)
+                    file_qmgr = open(fpath)
+                    cmd = [qmgr]
+                    d = self.du.run_cmd(
+                        self.hostname, cmd, stdin=file_qmgr, sudo=True,
+                        logerr=False, level=logging.DEBUG)
+                    file_qmgr.close()
+                # Load pbs.conf file
+                elif k == "pbs_conf":
+                    cfg_vals = base64.decodestring(v)
+                    config = ast.literal_eval(cfg_vals)
+                    self.du.set_pbs_config(self.hostname, confs=config)
+                # Load hooks
+                elif k == "hooks":
+                    fpath = self.du.create_temp_file()
+                    print_hooks = '\n'.join(v)
+                    with open(fpath, 'w') as f:
+                        f.write(print_hooks)
+                    file_qmgr = open(fpath)
+                    cmd = [qmgr]
+                    d = self.du.run_cmd(
+                        self.hostname, cmd, stdin=file_qmgr, sudo=True,
+                        logerr=False, level=logging.DEBUG)
+                    file_qmgr.close()
+            nodes = conf['pbsnodes']
+            for node in nodes:
+                node_name = str(node['id'])
+                self.create_pbsnode(node_name, node)
             return True
         return False
 
+    def create_pbsnode(self, node_name, attrs):
+        """
+        Create node in PBS with given attributes
```

Comment:
I have a doubt here, Suppose the server has a pre-config of one node and some vnodes of that execution host. Will this API `create_pbsnode` take care of that?

Reply:
I forgot to check for this condition, but here vnodes will not get created back. The reason is that I am deleting vnodedef file in teardown. I can add the save of vnodedef file in save configurations and load it back in teardown. But i feel that vnodedef file should be under save & load of Mom, since the scope of this PR is limited only to server.
What do you think @sujatapatnaik52 @hirenvadalia @agrawalravi90

Reply:
I am ok with handling vnodedefs with --use-current-setup in separate PR for MoM object

Reply:
I do agree with you. However, we should mention this in the EDD probably, if there are vnodes associated to the nodes then this option is not going load them back.

Reply:
I'm ok doing this in a different PR as well. I don't think the EDD needs to be modified, an EDD can be implemented via multiple PRs. Maybe instead make a note of it in the PR description?